sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2375,8 +2375,8 @@ class Dataset[T] private[sql](
     val cnvtr = converter.getOrElse(new ArrowConverters)
     withNewExecutionId {
       try {
-        val collectedRows = queryExecution.executedPlan.executeCollect()
-        cnvtr.internalRowsToPayload(collectedRows, this.schema)
+        val rowIter = queryExecution.executedPlan.executeToIterator()
```

Comment:
I am not sure this is what we want. I think executeToIterator computes each partition sequentially instead of in parrellel. 

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala#L287

https://github.com/apache/spark/blob/47d5d0ddb06c7d2c86515d9556c41dc80081f560/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L948